compiler/damlc/daml-stdlib-src/DA/Internal/Prelude.daml

```diff
@@ -445,6 +445,16 @@ span p xs@(x::xs')
          | p x          =  let (ys,zs) = span p xs' in (x::ys,zs)
          | otherwise    =  ([],xs)
 
+-- | Splits the given list into a pair of lists. All elements of the first list
```

Comment:
```suggestion
-- | The `partition` function takes a predicate a list and returns
-- the pair of lists of elements which do and do not satisfy the
-- predicate, respectively; i.e.,
--
-- > partition p xs == (filter p xs, filter (not . p) xs)
--
-- ```
-- >>> partition (`elem` "aeiou") "Hello World!"
-- ("eoo","Hll Wrld!")
-- ```
```
This is just the description from the Haskell standard library which I quite like. The translation to `filter` makes the behavior clear and it has an doctest in it which we test automatically.